sycl/source/detail/scheduler/scheduler.hpp

```diff
@@ -189,18 +190,21 @@ using ContextImplPtr = std::shared_ptr<detail::context_impl>;
 ///
 /// \ingroup sycl_graph
 struct MemObjRecord {
-  MemObjRecord(ContextImplPtr CurContext, std::size_t LeafLimit)
-      : MReadLeaves{LeafLimit}, MWriteLeaves{LeafLimit}, MCurContext{
-                                                             CurContext} {}
+  MemObjRecord(ContextImplPtr CurContext, std::size_t LeafLimit,
+               CircularBufferExtended::IfGenericIsFullF IfGenericIsFull,
+               CircularBufferExtended::AllocateDependencyF AllocateDependency)
+      : MReadLeaves{LeafLimit, IfGenericIsFull, AllocateDependency},
```

Comment:
Should we pass a pointer/reference to memonjrecord as host to eliminate passing of the pointer to push_back method?

Reply:
Considering the container is customized towards tracking leaves already, I think that change makes sense.

Reply:
Done.